src/common/snippets/src/lowered/pass/set_buffer_reg_group.cpp

```diff
@@ -113,47 +111,33 @@ std::vector<bool> SetBufferRegGroup::create_adjacency_matrix(LinearIR::constExpr
-    const auto input_count = loop_end->get_input_num();
-    const auto output_count = loop_end->get_output_num();
-
-    const auto& ptr_increments = loop_end->get_ptr_increments();
-    const auto& finalization_offsets = loop_end->get_finalization_offsets();
-    const auto& data_sizes = loop_end->get_element_type_sizes();
-
+SetBufferRegGroup::BufferMap SetBufferRegGroup::get_buffer_loop_neighbours(const UnifiedLoopInfoPtr& loop_info) {
     BufferMap buffer_neighbours;
-    for (size_t i = 0; i < input_count; ++i) {
-        const auto& parent_output = loop_end_expr->get_input_port_connector(i)->get_source().get_expr();
+
+    const auto& loop_inputs = loop_info->get_input_ports_info();
+    for (const auto& port_info : loop_inputs) {
+        const auto& parent_output = port_info.port.expr_port->get_port_connector_ptr()->get_source().get_expr();
         if (const auto buffer_expr = ov::as_type_ptr<BufferExpression>(parent_output)) {
             if (buffer_neighbours.count(buffer_expr) > 0) {
-                OPENVINO_ASSERT(buffer_neighbours[buffer_expr].ptr_increment == ptr_increments[i] &&
-                                buffer_neighbours[buffer_expr].finalization_offset == finalization_offsets[i],
+                const auto& port_desc = port_info.desc;
+                OPENVINO_ASSERT(buffer_neighbours[buffer_expr].desc == port_desc,
                                 "Invalid data pointer shifts: If Buffer has several consumers, this consumers must have the same shifts or zero");
                 continue;
             }
-            buffer_neighbours[buffer_expr] = { data_sizes[i], ptr_increments[i], finalization_offsets[i] };
+            buffer_neighbours[buffer_expr] = port_info;
         }
     }
-    for (size_t i = input_count; i < input_count + output_count; ++i) {
-        // The consumers of the corresponding Store ops
-        const auto consumer_inputs = loop_end_expr->get_input_port_connector(i)->get_consumers();
-        size_t buffer_count = 0;
-        size_t loop_count = 0;
+
+    const auto& loop_outputs = loop_info->get_output_ports_info();
+    for (const auto& port_info : loop_outputs) {
+        const auto& consumer_inputs = port_info.port.expr_port->get_port_connector_ptr()->get_consumers();
         for (const auto& consumer_input : consumer_inputs) {
             const auto& child_expr = consumer_input.get_expr();
-            if (const auto buffer_expr = ov::as_type_ptr<BufferExpression>(child_expr)) {
-                buffer_neighbours[buffer_expr] = { data_sizes[i], ptr_increments[i], finalization_offsets[i] };
-                buffer_count++;
-            } else if (ov::is_type<op::LoopEnd>(child_expr->get_node())) {
-                loop_count++;
-            }
-        }
-        if (buffer_count > 0) {
-            OPENVINO_ASSERT((buffer_count == 1) && (buffer_count + loop_count == consumer_inputs.size()),
-                            "Loop output must have not more than 1 Buffer");
```

Comment:
This check [exists](https://github.com/openvinotoolkit/openvino/blob/02176156f9f51127ffad5e259c0bad4573366ca6/src/common/snippets/src/lowered/pass/validate.cpp#L74) in `Validate` pass. Because of that I decided to remove it from this pass.